pilot/pkg/model/context.go

```diff
@@ -326,6 +326,8 @@ type WatchedResource struct {
 	// For Delta Xds, all resources of the TypeUrl that a client has subscribed to.
 	ResourceNames []string
 
+	CacheKeys map[string]XdsCacheEntry
```

Comment:
I think this could add great memory cost to istiod.  Last year we did a bunch of large scale performance test, and figured out that even the watched resource names of eds cost too much memory from pprof. The reason is even the eds request resources may have names ["cluster1", "cluster2"] same value as istiod inner generated one. But the string is unmarshalled from DiscoveryRequest, so in a process stack, they are not shared.  So I am not sure about this.

Reply:
I had similar concerns actually. I will do some benchmarking to see what cost there is.

It's possible we can keep around not the full key but just enough to do the DependsOn part

Reply:
One more concern, for eds the key is same as cds,